src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -14,68 +13,42 @@
 namespace Microsoft.CodeAnalysis.RemoveUnusedMembers
 {
     internal abstract class AbstractRemoveUnusedMembersDiagnosticAnalyzer<TDocumentationCommentTriviaSyntax, TIdentifierNameSyntax>
-        : AbstractCodeStyleDiagnosticAnalyzer
+        : DiagnosticAnalyzer, IBuiltInAnalyzer
```

Comment:
:bulb: Avoid using obsolete interface `IBuiltInAnalyzer`

Reply:
[IBuiltInAnalyzer](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/IBuiltInAnalyzer.cs,16) doesn't seem to be Obsolete, and I also see references to it in. If it is obsolete, when we should delete it as it is internal. Can you clarify?

Reply:
Removing it is a work in progress, starting with #26758. It will help the process by avoiding further use of it in new locations.

Reply:
Sounds reasonable, though I am worried if there will be any performance implications if we don't implement it. @heejaechang?

Reply:
Sure, but all the third party analyzers are not in the box by default, and can be uninstalled. Regardless, I don't think it will be a problem for this analyzer, as I believe `IBuiltInAnalyzer` is primarily needed to opt out of closed file analysis and/or reduce analysis scope for method body edits (which is not applicable here). I am going to follow your suggestion here, unless Heejae recommends otherwise.

Reply:
I dont think it being in the work to be removed can be a reason to block anyone from using it until it is actually removed .

I dont know when it is going to happen but, 
to remove OpenFileOnly, one must address simplify xxxx analyzer perf issue first in whatever way and make sure all analyzers can run on OOP.
to category, one must address whether the perf optimization this gives is worth enough to make it public or just remove it.

also, replacing OpenFileOnly to Hidden + NotConfigurable seems more confusing than OpenFileOnly. I would rather just expose another Tag which says OpenFileOnly.

Reply:
anyway, point is until IBuiltInAnalyzer is removed or has ETA when that will happen with workaround on how to replace each ability it is exposing, it seems fine to use until then.

Reply:
there is one more thing IBuiltInAnalyzer do. it lets us to report telemetry. if we are not using any of its ability then sure you don't need it but make sure you have Telemetry tag for all your diagnostics.